flink-connector-aws/flink-connector-aws-kinesis-streams/pom.xml

```diff
@@ -52,6 +60,11 @@ under the License.
             <artifactId>kinesis</artifactId>
         </dependency>
 
+        <dependency>
+            <groupId>software.amazon.kinesis</groupId>
+            <artifactId>amazon-kinesis-client</artifactId>
```

Comment:
This is a shame, can we do this without KCL? It's a heavy weight dependency for the 1 class(?) needed for deaggregation. We have been talking about removing this dependency with the new connector for years, but never really solved this bit. Can we ask Kinesis to publish a separate library for this? Or, we could just copy that file (assuming it is still a small asking of code)

Reply:
@dannycranmer  `KinesisClientRecord` has some transitive dependencies.
However, the KCL can be substantially lightened removing most of transitive dependencies
See [this example](https://github.com/aws-samples/amazon-managed-service-for-apache-flink-examples/blob/5e4873883c64c28484261125b3a2e20df3455c84/java/KinesisSourceDeaggregation/flink-app/pom.xml#L85) which is an alternative solution, implementing de-aggregation in the Deserialization Schema